pandas/core/dtypes/common.py

```diff
@@ -881,7 +882,9 @@ def is_datetime64_any_dtype(arr_or_dtype) -> bool:
     """
     if isinstance(arr_or_dtype, (np.dtype, ExtensionDtype)):
         # GH#33400 fastpath for dtype object
-        return arr_or_dtype.kind == "M"
+        if isinstance(arr_or_dtype, np.dtype):
+            return arr_or_dtype.kind == "M"
+        return arr_or_dtype.kind == "M" and arr_or_dtype.type != datetime.date
```

Comment:
It's not ideal that we're checking `datetime.date`. Thinking about it again. I guess I would be okay for pyarrow dates to return True if we don't need this work around.

FWIW, I don't think this method has any usage within pandas so I am not sure of the implication of changing this to `True`, but at least for pandas we want to avoid creep in logic checking

Reply:
I think it would be a poor experience for library developers to have errors like the following:
```python
import pandas as pd

s = pd.Series(['2000-01-01'], dtype='date32[pyarrow]')
if pd.api.types.is_datetime64_any_dtype(s):
    second = s.dt.second()
# ArrowNotImplementedError: Function 'second' has no kernel matching input types (date32[day])
```

But on the other hand the datetime64 refers to numpy's datetime64, right? So it's ok for it to return `True` if you're looking to cast it, like in my previous comment:
```python
import pyarrow as pa

type(pa.date32().to_pandas_dtype())
# numpy.dtypes.DateTime64DType 
type(pa.date64().to_pandas_dtype())
# numpy.dtypes.DateTime64DType
```

So I'm ok with what you decide. Should I restore it to return `True` for date types?

Reply:
Yeah I am leaning on returning `True` again. Sorry for the churn. 

In the use case you demonstrated, I suppose the type is still "datetime" in the fact that it supports the `dt` accessor but not so that Arrow implements `second` for `date32/64` specifically